### PR TITLE
"api-doc-entity-management"

### PR DESCRIPTION
**Pull request by MallanagoudaB**

### CODE REVIEW

File: src/api-doc/api-doc.yaml [hunk 1]
```diff
@@ -0,0 +1,2489 @@
...
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        _id:
+                          type: string
+                        entityType:
+                          type: string
+              examples:
+                example1:
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:2470]: @MallanagoudaB SuccessResponse

File: src/api-doc/api-doc.yaml [hunk 2]
```diff
@@ -0,0 +1,2489 @@
...
+                    result:
+                      - _id: 66738b75c05aa58f89ba1300
+                        entityType: block
+        '400':
+          description: 'Bad Request. '
+          content:
+            application.json:
+              schema: null
+              examples:
+                example1:
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:2483]: @MallanagoudaB FailureResponse

File: src/api-doc/api-doc.yaml [hunk 3]
```diff
@@ -0,0 +1,2489 @@
...
+    post:
+      summary: Cancel enrollment from a session
+      tags:
+        - entities-API's
+      description: >-
+        You can use this API to map the childEntity to parentEntity
+
+
+        * The API Endpoint for cancelling enrollment for a session is
+        `/v1/entities/mappingUpload`
```
[the remark this answers is not in the document]
Comment (MallanagoudaB) [reply src/api-doc/api-doc.yaml:1607]: no params we only require csv file in body

File: src/api-doc/api-doc.yaml [hunk 4]
```diff
@@ -0,0 +1,2165 @@
...
+    post:
+      operationId: ''
+      summary: This api will reate's Single entity Type data
+      tags:
+        - entityType-API's
+      description: >-
+        You can use this API to create a single entityType data.
+
+
+        * The API Endpoint to read form data is `/v1/entityTypes/create/`
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:186]: @MallanagoudaB Why there is / after create in URL?

File: src/api-doc/api-doc.yaml [hunk 5]
```diff
@@ -0,0 +1,2165 @@
...
+                FailureResponse:
+                  value:
+                    status: 400
+                    message:
+                      - location: body
+                        param: name
+                        msg: required name
+  /v1/entityTypes/update/{_id}:
+    post:
+      summary: Update single entities based on the Id's
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:283]: @MallanagoudaB based on what IDs?

File: src/api-doc/api-doc.yaml [hunk 6]
```diff
@@ -0,0 +1,2489 @@
...
+                          type: string
+                        name:
+                          type: string
+                      required:
+                        - _id
+                        - name
+              examples:
+                example1:
+                  value:
+                    message: ENTITY_TYPES_FETCHED
```
[the remark this answers is not in the document]
Comment (VISHNUDAS-tunerlabs) [reply src/api-doc/api-doc.yaml:504]: @MallanagoudaB Why did you resolve this comment without actually resolving it?

File: src/api-doc/api-doc.yaml [hunk 7]
```diff
@@ -0,0 +1,2165 @@
...
+              examples:
+                SuccessResponse:
+                  value:
+                    message: ENTITY_TYPES_FETCHED
+                    status: 200
+                    result:
+                      - _id: 661384681797bc00de520555
+                        name: school
+                      - _id: 661387f572fad002e57cce8f
+                        name: school
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:510]: @MallanagoudaB change this to some other entity type name

File: src/api-doc/api-doc.yaml [hunk 8]
```diff
@@ -0,0 +1,2165 @@
...
+                      - location: body
+                        param: entities
+                        msg: entities file is required
+  /v1/entities/bulkUpdate:
+    post:
+      summary: Bulk updates of entities
+      tags:
+        - entities-API's
+      description: >-
+        You can use this API to entities data based on the type and ID we pass
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:714]: @MallanagoudaB what id?

File: src/api-doc/api-doc.yaml [hunk 9]
```diff
@@ -0,0 +1,2165 @@
...
+                    - beat
+                    - cluster
+                    - school
+                    - block
+                  allowedRoles:
+                    - role1
+                    - role2
+  /v1/entities/details/{_id}:
+    get:
+      summary: Provide details of an entities based on entitie ID.
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:963]: @MallanagoudaB spelling mistake

File: src/api-doc/api-doc.yaml [hunk 10]
```diff
@@ -0,0 +1,2165 @@
...
+
+        * It is mandatory to provide values for parameters marked with
+        `required`
+
+        * Mandatory parameter cannot be empty or null.
+      parameters:
+        - in: path
+          name: _id
+          required: true
+          description: Please append a valid entitie id to the Request URL..
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:980]: @MallanagoudaB remove one full stop

File: src/api-doc/api-doc.yaml [hunk 11]
```diff
@@ -0,0 +1,2165 @@
...
+          description: >-
+            You require an X-auth-token to use this API., which is available in
+            the login API Response.
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: _id
+          required: true
+          description: Please append a valid entities Id to the Request URL..
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:1094]: @MallanagoudaB remove extra full stop

File: src/api-doc/api-doc.yaml [hunk 12]
```diff
@@ -0,0 +1,2165 @@
...
+                FailureResponse:
+                  value:
+                    status: 400
+                    message:
+                      - location: body
+                        param: entityMap
+                        msg: entityMap file is required
+  /v1/entities/listByEntityType/{_id}:
+    get:
+      summary: You can use this API to list entities based on its type
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:1290]: @MallanagoudaB make type to entity type

File: src/api-doc/api-doc.yaml [hunk 13]
```diff
@@ -0,0 +1,2165 @@
...
+  /v1/entities/relatedEntities/{_id}:
+    get:
+      summary: get related entities based on ID
+      tags:
+        - entities-API's
+      description: >-
+        You can use this API to get the realted entities data based on ID
+
+
+        * The API Endpoint to get the realted entities data based on ID:
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:1593]: @MallanagoudaB still spelling issue why this comment was resolved?

File: src/api-doc/api-doc.yaml [hunk 14]
```diff
@@ -0,0 +1,2489 @@
...
+                      - location: params
+                        param: _id
+                        msg: required entity id
+  /v1/entities/relatedEntities/_id:
+    get:
+      summary: get related entities based on ID
+      tags:
+        - entities-API's
+      description: >-
+        You can use this API to get the realted entities data based on ID
```
[the remark this answers is not in the document]
Comment (VISHNUDAS-tunerlabs) [reply src/api-doc/api-doc.yaml:1590]: @MallanagoudaB you resolved without correcting the spelling

File: src/api-doc/api-doc.yaml [hunk 15]
```diff
@@ -0,0 +1,2165 @@
...
+                          required:
+                            - _id
+                            - entityType
+                            - externalId
+                            - label
+                            - value
+                      count:
+                        type: number
+              examples:
+                Success Response:
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:2009]: @MallanagoudaB use same key every where

File: src/api-doc/Entity-Management-API's.json [hunk 16]
```diff
@@ -0,0 +1,1046 @@
+{
+	"info": {
+		"_postman_id": "fa7d1558-cdcb-493a-8767-55758ac735da",
+		"name": "Entity Management API's",
+		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
+		"_exporter_id": "33513401"
+	},
+	"item": [
+		{
+			"name": "EntityType Service",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:10]: @MallanagoudaB why entityType service?

File: src/api-doc/Entity-Management-API's.json [hunk 17]
```diff
@@ -0,0 +1,1046 @@
...
+							"mode": "raw",
+							"raw": "{\n    \"query\": {\n        \"name\": \"Goa\"\n    },\n    \"projection\": {\n        \"immediateChildrenEntityType\": 1,\n        \"name\": 1\n    },\n    \"skipFields\": {\n        \"createdAt\": 1\n    }\n}\n",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entityTypes/find",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:43]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 18]
```diff
@@ -0,0 +1,1046 @@
...
+							"mode": "raw",
+							"raw": "{\n    \"name\": \"state\",\n    \"registryDetails\": {\n        \"name\": \"schoolRegistry\"\n    },\n    \"isObservable\": true,\n    \"toBeMappedToParentEntities\": true\n}\n",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entityTypes/create",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:138]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 19]
```diff
@@ -0,0 +1,1046 @@
...
+							"formdata": [
+								{
+									"key": "entityTypes",
+									"type": "file",
+									"src": "/home/user3/Documents/EntityType-Upload_2024_05_07_09_21.csv"
+								}
+							]
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entityTypes/bulkUpdate",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:187]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 20]
```diff
@@ -0,0 +1,1046 @@
...
+								"type": "text"
+							},
+							{
+								"key": "internal-access-token",
+								"value": "8c3a94f0931e01a4940a",
+								"type": "text"
+							}
+						],
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entityTypes/list",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:225]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 21]
```diff
@@ -0,0 +1,1046 @@
...
+								"type": "text"
+							},
+							{
+								"key": "content-type",
+								"value": "multipart/form-data",
+								"type": "text"
+							},
+							{
+								"key": "x-authenticated-token",
+								"value": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhIjp7ImlkIjoxLCJuYW1lIjoiUHJpeWFua2EiLCJzZXNzaW9uX2lkIjoyMjcsIm9yZ2FuaXphdGlvbl9pZCI6MSwicm9sZXMiOlt7ImlkIjo3LCJ0aXRsZSI6Im9yZ19hZG1pbiIsInVzZXJfdHlwZSI6MSwic3RhdHVzIjoiQUNUSVZFIiwib3JnYW5pemF0aW9uX2lkIjoxLCJ2aXNpYmlsaXR5IjoiUFVCTElDIn0seyJpZCI6NSwidGl0bGUiOiJtZW50ZWUiLCJ1c2VyX3R5cGUiOjAsInN0YXR1cyI6IkFDVElWRSIsIm9yZ2FuaXphdGlvbl9pZCI6MSwidmlzaWJpbGl0eSI6IlBVQkxJQyJ9LHsiaWQiOjgsInRpdGxlIjoicHJvZ3JhbV9kZXNpZ2luZXIiLCJ1c2VyX3R5cGUiOjAsInN0YXR1cyI6IkFDVElWRSIsIm9yZ2FuaXphdGlvbl9pZCI6MSwidmlzaWJpbGl0eSI6IlBVQkxJQyJ9LHsiaWQiOjksInRpdGxlIjoicHJvZ3JhbV9jcmVhdG9yIiwidXNlcl90eXBlIjowLCJzdGF0dXMiOiJBQ1RJVkUiLCJvcmdhbml6YXRpb25faWQiOjEsInZpc2liaWxpdHkiOiJQVUJMSUMifV19LCJpYXQiOjE3MTkyMTkzNDksImV4cCI6MTcxOTIyMTE0OX0.ZGhNl3g69zlTtRwCseq0PkXgBqTbf-e0syRYV2KXchI",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:258]: @MallanagoudaB do not append actual token.

File: src/api-doc/Entity-Management-API's.json [hunk 22]
```diff
@@ -0,0 +1,1046 @@
...
+								{
+									"key": "",
+									"type": "file",
+									"src": [],
+									"disabled": true
+								}
+							]
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entityTypes/bulkCreate",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:279]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 23]
```diff
@@ -0,0 +1,1046 @@
...
+								"bulkCreate"
+							]
+						}
+					},
+					"response": []
+				}
+			]
+		},
+		{
+			"name": "Entities Service",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:298]: @MallanagoudaB why this is entities service? do we have service like that! make it entities

File: src/api-doc/Entity-Management-API's.json [hunk 24]
```diff
@@ -0,0 +1,1046 @@
...
+								"type": "text"
+							},
+							{
+								"key": "content-type",
+								"value": "application/json",
+								"type": "text"
+							}
+						],
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/listByEntityType/5f32d8228e0dc8312404056e?page=1",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:322]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 25]
```diff
@@ -0,0 +1,1046 @@
...
+							"mode": "raw",
+							"raw": "{\n    \"externalId\": \"entity123\",\n     \"name\": \" block\",\n    \"childHierarchyPath\": [],\n    \"allowedRoles\": [\"role2\", \"role1\"],\n    \"createdBy\": \"user123\",\n    \"updatedBy\": \"user123\"\n}\n",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/update/663364443c990eaa179e289e",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:376]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 26]
```diff
@@ -0,0 +1,1046 @@
...
+								{
+									"key": "entityType",
+									"type": "file",
+									"src": [],
+									"disabled": true
+								}
+							]
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/bulkCreate?type=school",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:431]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 27]
```diff
@@ -0,0 +1,1046 @@
...
+							"formdata": [
+								{
+									"key": "entities",
+									"type": "file",
+									"src": "/home/user3/Music/Entity-Upload_2024_04_08_14_59.csv"
+								}
+							]
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/bulkUpdate",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:486]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 28]
```diff
@@ -0,0 +1,1046 @@
...
+							"formdata": [
+								{
+									"key": "entityMap",
+									"type": "file",
+									"src": "/home/user3/Videos/mapping.csv"
+								}
+							]
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/mappingUpload",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:534]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 29]
```diff
@@ -0,0 +1,1046 @@
...
+								"type": "text"
+							},
+							{
+								"key": "x-authenticated-token",
+								"value": "{{token}}",
+								"type": "text"
+							}
+						],
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/list/5f33c3d85f637784791cd830?page=1&limit=100&type=school",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:572]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 30]
```diff
@@ -0,0 +1,1046 @@
...
+							"mode": "raw",
+							"raw": "",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/relatedEntities/6153158e5bec1d28882358ab",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:637]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 31]
```diff
@@ -0,0 +1,1046 @@
...
+							"mode": "raw",
+							"raw": "{ \"query\": {\n        \"_id\": {\n            \"$in\": [\"5fd1b52ab53a6416aaeefb1f\", \"5fd1b52ab53a6416aaeefb24\",\"5f33c3d85f637784791cd830\",\"5fd1b52ab53a6416aaeefb26\",\"5fd098e2e049735a86b748ae\"]\n        }\n    },\n    \"projection\": [\n        \"_id\",\"metaInformation.name\"\n    ]}",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/find",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:686]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 32]
```diff
@@ -0,0 +1,1046 @@
...
+								"type": "text"
+							},
+							{
+								"key": "x-authenticated-token",
+								"value": "{{token}}",
+								"type": "text"
+							}
+						],
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/details/663339bc0cb19f01c459853b",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:724]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 33]
```diff
@@ -0,0 +1,1046 @@
...
+							"raw": "{\n  \"metaInformation.name\": [\n    \"Maharashtra\"\n  ]\n}\n",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/listByLocationIds",
+							"protocol": "http",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:776]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 34]
```diff
@@ -0,0 +1,1046 @@
...
+							"mode": "raw",
+							"raw": "{\n    \"entityTypeId\": \"663335e2a766490014aead0a\",\n    \"externalId\": \"entity44123\",\n    \"name\": \"Arunachal Pradesh\",\n     \"childHierarchyPath\" : [ \n        \"district\", \n        \"beat\", \n        \"cluster\", \n        \"school\", \n        \"block\"\n    ],\n    \"allowedRoles\": [\"role1\", \"role2\"]\n    \n}\n",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/add?type=state",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:822]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 35]
```diff
@@ -0,0 +1,1046 @@
...
+							},
+							{
+								"key": "internal-access-token",
+								"value": "8c3a94f0931e01a4940a",
+								"type": "text",
+								"disabled": true
+							}
+						],
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/subEntityListBasedOnRoleAndLocation/entity123",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:867]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 36]
```diff
@@ -0,0 +1,1046 @@
...
+							},
+							{
+								"key": "internal-access-token",
+								"value": "8c3a94f0931e01a4940a",
+								"type": "text",
+								"disabled": true
+							}
+						],
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/subEntityList/663354e56109810671aec49f?type=school&search=&page=1&limit=100",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:914]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 37]
```diff
@@ -0,0 +1,1046 @@
...
+							"mode": "raw",
+							"raw": "{\n    \"entities\": [\n        \"5f33c3d85f637784791cd830\"\n    ],\n    \"fields\": [\n        \"entityType\"\n    ]\n}\n",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/listByIds",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:984]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

File: src/api-doc/Entity-Management-API's.json [hunk 38]
```diff
@@ -0,0 +1,1046 @@
...
+								"disabled": true
+							},
+							{
+								"key": "x-authenticated-token",
+								"value": "{{userToken}}",
+								"type": "text"
+							}
+						],
+						"url": {
+							"raw": "http://localhost:5001/entity/v1/entities/entityListBasedOnEntityType?entityType=state",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity-Management-API's.json:1021]: @MallanagoudaB instead http://localhost:5001 give variable from environment eg: {{variable}}

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed on jul-10

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 39]
```diff
@@ -0,0 +1,977 @@
+{
+	"info": {
+		"_postman_id": "fa7d1558-cdcb-493a-8767-55758ac735da",
+		"name": "Entity Management API's",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:4]: @MallanagoudaB Lets keep it as Entity-Management

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 40]
```diff
@@ -0,0 +1,977 @@
...
+		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
+		"_exporter_id": "33513401"
+	},
+	"item": [
+		{
+			"name": "entity",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:10]: @MallanagoudaB it is not entity it is entityType

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 41]
```diff
@@ -0,0 +1,977 @@
...
+							"mode": "raw",
+							"raw": "{\n    \"query\": {\n        \"name\": \"Goa\"\n    },\n    \"projection\": {\n        \"immediateChildrenEntityType\": 1,\n        \"name\": 1\n    },\n    \"skipFields\": {\n        \"createdAt\": 1\n    }\n}\n",
+							"options": {
+								"raw": {
+									"language": "json"
+								}
+							}
+						},
+						"url": {
+							"raw": "{{baseUrl}}/entity/v1/entityTypes/find",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:43]: @MallanagoudaB {{baseUrl}}/entity/ will not work right it should be {{baseUrl}}/entity-management/
Please check all the urls in this postman collection. Very Important! not repeating same comment  again again

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 42]
```diff
@@ -0,0 +1,977 @@
...
+		"name": "Entity Management API's",
+		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
+		"_exporter_id": "33513401"
+	},
+	"item": [
+		{
+			"name": "entity",
+			"item": [
+				{
+					"name": "Find",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:13]: @MallanagoudaB use same convention of endpoint , means make it "find". all lower case letters

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 43]
```diff
@@ -0,0 +1,977 @@
...
+								"v1",
+								"entityTypes",
+								"find"
+							]
+						}
+					},
+					"response": []
+				},
+				{
+					"name": "Update",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:58]: @MallanagoudaB all lowercase letters matching the endpoint

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 44]
```diff
@@ -0,0 +1,977 @@
...
+								"entityTypes",
+								"update",
+								"663335e2a766490014aead0a"
+							]
+						}
+					},
+					"response": []
+				},
+				{
+					"name": "Create",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:104]: @MallanagoudaB all lowercase letters matching the endpoint

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 45]
```diff
@@ -0,0 +1,977 @@
...
+									"key": "page",
+									"value": "1"
+								}
+							]
+						}
+					},
+					"response": []
+				},
+				{
+					"name": "Update",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:331]: @MallanagoudaB all lowercase letters matching the endpoint

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 46]
```diff
@@ -0,0 +1,977 @@
...
+								"entities",
+								"update",
+								"663364443c990eaa179e289e"
+							]
+						}
+					},
+					"response": []
+				},
+				{
+					"name": "BulkCreate",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:377]: @MallanagoudaB all lowercase letters matching the endpoint

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 47]
```diff
@@ -0,0 +1,977 @@
...
+									"key": "type",
+									"value": "school"
+								}
+							]
+						}
+					},
+					"response": []
+				},
+				{
+					"name": "BulkUpdate",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:435]: @MallanagoudaB all lowercase letters matching the endpoint

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 48]
```diff
@@ -0,0 +1,977 @@
...
+								"v1",
+								"entities",
+								"find"
+							]
+						}
+					},
+					"response": []
+				},
+				{
+					"name": "Details",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:667]: @MallanagoudaB all lowercase letters matching the endpoint

File: src/api-doc/Entity Management API's.postman_collection.json [hunk 49]
```diff
@@ -0,0 +1,977 @@
...
+								"v1",
+								"entities",
+								"listByLocationIds"
+							]
+						}
+					},
+					"response": []
+				},
+				{
+					"name": "Add",
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/Entity Management API's.postman_collection.json:752]: @MallanagoudaB all lowercase letters matching the endpoint

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
reviewed on jul 12-2024 3:50 PM

File: src/api-doc/api-doc.yaml [hunk 50]
```diff
@@ -0,0 +1,2220 @@
...
+                value:
+                  locationIds:
+                    - entity44123
+  /v1/entities/subEntityListBasedOnRoleAndLocation/{_id}:
+    get:
+      summary: list all the entities based on location id
+      tags: &ref_5
+        - entities-API's
+      description: >-
+        You can use this API to list entity based on location id
```
[the remark this answers is not in the document]
Comment (MallanagoudaB) [reply src/api-doc/api-doc.yaml:1882]: no @VISHNUDAS-tunerlabs  only locationid both location and code have same value

File: src/api-doc/api-doc.yaml [hunk 51]
```diff
@@ -332,7 +332,7 @@ paths:
           required: true
         - in: query
           name: _id
-          description: ''
+          description: 'to update the data we need to match its id'
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:335]: @MallanagoudaB bad description please add a proper one

File: src/api-doc/api-doc.yaml [hunk 52]
```diff
@@ -679,7 +680,7 @@ paths:
             type: string
         - in: query
           name: type
-          description: type we need to pass to match the entityType
+          description: entity type of the entities we need to pass to match the entityType
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:683]: @MallanagoudaB bad description :  
Please pass entity type value for the entity, 
check for similar changes

File: src/api-doc/api-doc.yaml [hunk 53]
```diff
@@ -1340,9 +1340,9 @@ paths:
                       - location: body
                         param: entityMap
                         msg: entityMap file is required
-  /v1/entities/listByEntityType/{_id}:
+  /v1/entities/listByEntityType/{_id}?page={page_no}:
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:1343]: @MallanagoudaB  where is limit?

File: src/api-doc/api-doc.yaml [hunk 54]
```diff
@@ -1372,6 +1372,12 @@ paths:
           schema:
             type: string
           example: 1
+        - in: path
+          name: page
+          description: Please appenda number for pagination.
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:1377]: @MallanagoudaB spelling mistake

File: src/api-doc/api-doc.yaml [hunk 55]
```diff
@@ -0,0 +1,2226 @@
...
+servers:
+  - url:  http://{{baseurl}}/entity-management
+    description: local server
+  - url: https://dev.elevate-apis.shikshalokam.org
+    description: dev server
+paths:
+  /v1/entityTypes/bulkCreate:
+    post:
+      operationId: ''
+      summary: 'bulkCreate create entityType '
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:20]: @MallanagoudaB wrong summary. it should be Bulk create entity type

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
13-jul-2014 12:17 AM

File: src/api-doc/api-doc.yaml (hunk 53 above)
Comment (MallanagoudaB) [reply src/api-doc/api-doc.yaml:1343]: we dont have limit for this api

File: src/api-doc/api-doc.yaml [hunk 56]
```diff
@@ -205,7 +205,7 @@ paths:
         You can use this API to create a single entityType data.
 
 
-        * The API Endpoint to read form data is `/v1/entityTypes/create`
+        * The API Endpoint tocreate a single entityType data is `/v1/entityTypes/create`
```
Comment (VISHNUDAS-tunerlabs) [inline src/api-doc/api-doc.yaml:208]: @MallanagoudaB space missing

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]

**Review by VISHNUDAS-tunerlabs — APPROVED** [review]

**Pull request merged**